tests/framework/extensions/provisioning/verify.go

```diff
@@ -450,3 +457,70 @@ func VerifySSHTests(t *testing.T, client *rancher.Client, clusterObject *steveV1
 	}
 
 }
+
+func CreateAndVerifyAWSLoadBalancer(t *testing.T, client *rancher.Client, clusterName string) {
+	adminClient, err := rancher.NewClient(client.RancherConfig.AdminToken, client.Session)
+	require.NoError(t, err)
+
+	steveclient, err := adminClient.Steve.ProxyDownstream(clusterName)
+	require.NoError(t, err)
+
+	nginxWorkload, err := workloads.CreatSimpleDeployment(steveclient, "nginx")
+	require.NoError(t, err)
+
+	nginxSpec := &appv1.DeploymentSpec{}
+	err = steveV1.ConvertToK8sType(nginxWorkload.Spec, nginxSpec)
+	require.NoError(t, err)
+
+	_, err = workloads.CreateService(steveclient, "nginx", 80, corev1.ServiceTypeClusterIP, nginxSpec.Selector.MatchLabels)
+	require.NoError(t, err)
+
+	serviceLB, err := workloads.CreateService(steveclient, "nginx-lb", 80, corev1.ServiceTypeLoadBalancer, nginxSpec.Selector.MatchLabels)
+	require.NoError(t, err)
+
+	lbHostname := ""
+	statusCode := 500
+	err = kwait.Poll(5*time.Second, 1*time.Minute, func() (done bool, err error) {
+		updateService, err := steveclient.SteveType("service").ByID(serviceLB.ID)
+		if err != nil {
+			return false, nil
+		}
+
+		serviceStatus := &corev1.ServiceStatus{}
+		err = steveV1.ConvertToK8sType(updateService.Status, serviceStatus)
+		if err != nil {
+			return false, err
+		}
+		if len(serviceStatus.LoadBalancer.Ingress) == 0 {
+			return false, nil
+		}
+
+		lbHostname = serviceStatus.LoadBalancer.Ingress[0].Hostname
+		return true, nil
+	})
+	require.NoError(t, err)
+
+	err = kwait.Poll(5*time.Second, 3*time.Minute, func() (done bool, err error) {
+		req, err := http.NewRequest("GET", "http://"+lbHostname, nil)
+		if err != nil {
+			return false, err
+		}
+		req.Header.Add("Authorization", "Bearer "+client.RancherConfig.AdminToken)
+		response, err := client.Management.APIBaseClient.Ops.Client.Do(req)
+		if err != nil {
+			// ignoring this error, LB may not be ready yet
+			return false, nil
+		}
+		if response == nil {
+			logrus.Info("response is nil")
+			return false, nil
+		}
+		if response.StatusCode == 200 {
+			statusCode = response.StatusCode
+			return true, nil
+		}
+		return false, nil
```

Comment:
So for this, there's already a raw HTTP request function. Here: https://github.com/rancher/rancher/blob/release/v2.8/tests/framework/extensions/charts/charts.go#L333. And it's used here underneath for the ingress: https://github.com/rancher/rancher/blob/release/v2.8/tests/framework/extensions/ingresses/ingresses.go#L12-L21.

What do you all think about moving charts' raw HTTP logic to a new extension in the extensions/services and calling something like _AccessExternally_ (placeholder)? This newly created extension could be used:

- Here within this poll
- Within `AccessIngressExternally`
- Within `GetChartCaseEndpoint`

Reply:
I think this is great idea except for the `AccessExternally` we should just have the `AccessIngressExternally` extension and move this logic there.

Reply:
I have not addressed this yet, will do this soon .

Reply:
would this need to be done as part of this PR, or separately?